planner/core/logical_plan_builder.go

```diff
@@ -1990,6 +1994,37 @@ func (b *PlanBuilder) buildDataSource(tn *ast.TableName) (LogicalPlan, error) {
 	return result, nil
 }
 
+func (b *PlanBuilder) buildDataSourceFromView(dbName model.CIStr, tableInfo *model.TableInfo) (LogicalPlan, error) {
+	var (
+		selectNode        ast.StmtNode
+		selectLogicalPlan Plan
+		err               error
+	)
+	charset, collation := b.ctx.GetSessionVars().GetCharsetInfo()
+	selectNode, err = parser.New().ParseOneStmt(tableInfo.View.SelectStmt, charset, collation)
+	if err != nil {
+		return nil, err
+	}
+	selectLogicalPlan, err = b.Build(selectNode)
+	if err != nil {
+		return nil, err
+	}
+	var viewCols []*expression.Column
+	for i := range tableInfo.View.Cols {
+		col := selectLogicalPlan.Schema().FindColumnByName(tableInfo.View.Cols[i].L)
+		if col == nil {
+			return nil, ErrViewInvalid.GenWithStackByArgs(dbName.L, tableInfo.Name.L)
+		}
+		col.ColName = tableInfo.Cols()[i].Name
+		col.OrigColName = tableInfo.View.Cols[i]
+		viewCols = append(viewCols, col)
+	}
+	projUponView := LogicalProjection{Exprs: expression.Column2Exprs(viewCols)}.Init(b.ctx)
+	projUponView.SetChildren(selectLogicalPlan.(LogicalPlan))
+	projUponView.SetSchema(expression.NewSchema(viewCols...))
```

Comment:
We should build schema.Columns instead of using the selectLogicalPlan.Schema.Columns

``` go
&expression.Column{
		UniqueID:    b.ctx.GetSessionVars().AllocPlanColumnID(),
		TblName:     viewname,
		OrigTblName: tableInfo.Cols[i].Name,
		ColName:     colName,
		OrigColName: tableInfo.View.Cols[i].Name,
		DBName:      dbName,
		RetType:     expr.GetType(),
	}
```

Reply:
so I just use line2018 and line2019 to replace ColName and OrigColName, all the other attribute almost the same value.

Reply:
We should not use the column in the child schema as the parent schema

Reply:
Understand, PTAL again